### ISSUE TITLE
Versioning scheme is already messed up

### ISSUE THREAD

**Issue by cloudishBenne** [mumble-voip/mumble-docker#19]
If anyone has a problem with the 1.4.230 container image, you now have to ask "which one do you mean, the old or the new one?"
A hotfix, patch or whatever should be the best reason there is semantic versioning.
Maybe start using a scheme like "X.Y.Z-N" so you can track the container changes in N. To differentiate bigger changes in how one can configure the container, you could use "X.Y.Z-N.O" or just "X.Y.Z-N" plus wait with the bigger changes till "X.Y.(Z+1)-1 gets a release.

But please, don't just replace the release.

Matching discussions:
https://github.com/semver/semver/issues/709
https://github.com/semver/semver/issues/703#issuecomment-841559625

**Comment by azlux** [comment]
You're right, the image have changes unrelated to mumble server release.
I think we will use the "Debian" version. v1.4.230~1 (X.Y.Z~N) , I like this one, It's quite simple.
I need some changes into the build process to allow this new format.

**Comment by cloudishBenne** [comment]
I think this is the right decision.
So the original first release would now be 1.4.230~1 and the now second release would be 1.4.230~2?
I would suggest you re-release the first one again too, even if the new naming scheme will be confusing. But the commit-hashes would match to the releases.

**Comment by Krzmbrzl** [comment]
Other possibilities:
- 1.4.230-1
- 1.4.230+1
- 1.4.230.1
- 1.4.230a
- 1.4.230#1
- 1.4.230*1

But yeah, `-` looks good to me :+1:

**Comment by azlux** [comment]
The commit https://github.com/mumble-voip/mumble-docker/commit/2ac6680a880288d20e71cbcdaad5b2b74303bf7b allow us to use X.Y.Z-N as tag/release name.  And support no sub-version at all (v1.4.230)

The `-` is the character between the mumble version and the container "sub-"release
@cloudishBenne not sure it's really useful to deploy old versions. Maybe deploying now a "v1.4.230-1" version.

Suggestion:
- Release a version when mumble have a new release with the same name.
- Release a version with the `-N` (starting at 1) when there are change on this repo independent of mumble server project.

**Comment by Krzmbrzl** [comment]
> Suggestion:
>
 >   Release a version when mumble have a new release with the same name.
 >   Release a version with the -N (starting at 1) when there are change on this repo independent of mumble server project.

Nah, I'd directly add the -N part to stay consistent within the used versioning format

**Comment by cloudishBenne** [comment]
I think always adding the -N is more consistent. This allows in future on dockerhub those fancy things with doubletagging or however this is called. So the generic 1.4.230 container tag would always point to the latest 1.4.230-N and this information would be transported everytime with every version-tag.

**Comment by fgma** [comment]
@cloudishBenne That would also lead to the same problem again. Which version 1.4.230 are you actually running when reporting a bug?

**Comment by cloudishBenne** [comment]
> @cloudishBenne That would also lead to the same problem again. Which version 1.4.230 are you actually running when reporting a bug?

No this is totally normal on dockerhub and just references which image to pull, not what the complete version of the image is.
Look e.g. here:
https://hub.docker.com/_/python/?tab=description

**Comment by cloudishBenne** [comment]
For a small server with standard config it usually is bs to pin to a specific release. If something blows up, you fix it by hand.
But if you manage a fleet of servers with automatic scaling, you probably have deployment scripts or tools that rely on a fixed behaviour. Or you have a little patchscript that fixes the issue why there was a rerelease here in the first place. Then you don't want to change because it works.
